File: testdata/jennies/builders/builder_delegation_in_disjunction/PHPBuilder/src/BuilderDelegationInDisjunction/DashboardBuilder.php
<?php

namespace Grafana\Foundation\BuilderDelegationInDisjunction;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\Dashboard>
 */
class DashboardBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BuilderDelegationInDisjunction\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BuilderDelegationInDisjunction\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BuilderDelegationInDisjunction\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * will be expanded to cog.Builder<DashboardLink> | string
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink>|string $singleLinkOrString
     */
    public function singleLinkOrString( $singleLinkOrString): static
    {
        /** @var \Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink|string $singleLinkOrStringResource */
        $singleLinkOrStringResource = $singleLinkOrString instanceof \Grafana\Foundation\Cog\Builder ? $singleLinkOrString->build() : $singleLinkOrString;
        $this->internal->singleLinkOrString = $singleLinkOrStringResource;
    
        return $this;
    }

    /**
     * will be expanded to [](cog.Builder<DashboardLink> | string)
     * @param array<\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink>|string> $linksOrStrings
     */
    public function linksOrStrings(array $linksOrStrings): static
    {
            $linksOrStringsResources = [];
            foreach ($linksOrStrings as $r1) {
                    $linksOrStringsResources[] = $r1 instanceof \Grafana\Foundation\Cog\Builder ? $r1->build() : $r1;
            }
        $this->internal->linksOrStrings = $linksOrStringsResources;
    
        return $this;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink>|\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\ExternalLink> $disjunctionOfBuilders
     */
    public function disjunctionOfBuilders( $disjunctionOfBuilders): static
    {
        $disjunctionOfBuildersResource = $disjunctionOfBuilders->build();
        $this->internal->disjunctionOfBuilders = $disjunctionOfBuildersResource;
    
        return $this;
    }

}

File: testdata/jennies/builders/builder_delegation_in_disjunction/CSharpBuilder/src/Grafana/Foundation/BuilderDelegationInDisjunction/DashboardBuilder.cs
namespace Grafana.Foundation.BuilderDelegationInDisjunction;


public class DashboardBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public DashboardBuilder()
    {
        this.@internal = new Dashboard();
    }

    public DashboardBuilder SingleLinkOrString(Cog.IBuilder<object> singleLinkOrString)
    {
        object singleLinkOrStringResource = singleLinkOrString.Build();
        this.@internal.SingleLinkOrString = singleLinkOrStringResource;
        return this;
    }

    public DashboardBuilder LinksOrStrings(List<Cog.IBuilder<object>> linksOrStrings)
    {
        List<object> linksOrStringsResources = new List<object>();
        foreach (Cog.IBuilder<object> r1 in linksOrStrings)
        {
                object linksOrStringsDepth1 = r1.Build();
                linksOrStringsResources.Add(linksOrStringsDepth1);
        }
        this.@internal.LinksOrStrings = linksOrStringsResources;
        return this;
    }

    public DashboardBuilder DisjunctionOfBuilders(Cog.IBuilder<object> disjunctionOfBuilders)
    {
        object disjunctionOfBuildersResource = disjunctionOfBuilders.Build();
        this.@internal.DisjunctionOfBuilders = disjunctionOfBuildersResource;
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
